File: telerik_ui_for_winforms_courseware_chm/Courseware/Grid/CS/RadGridView/CustomCells/ColorSwatchColumn.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Telerik.WinControls.UI;

namespace _15_CustomCells
{
    class ColorSwatchColumn : GridViewDataColumn
    {
        public ColorSwatchColumn(string fieldName)
            : base(fieldName)
        {

        }

        public override Type GetCellType(GridViewRowInfo row)
        {
            if (row is GridViewDataRowInfo)
            {
                return typeof(ColorSwatchCellElement);
            }
            return base.GetCellType(row);
        }
    }
}

File: telerik_ui_for_winforms_courseware_chm/Courseware/Grid/VB/RadGridView/CustomCells/ColorSwatchColumn.vb
Imports Telerik.WinControls.UI

Class ColorSwatchColumn
    Inherits GridViewDataColumn
    Public Sub New(ByVal fieldName As String)

        MyBase.New(fieldName)
    End Sub

    Public Overrides Function GetCellType(ByVal row As GridViewRowInfo) As System.Type
        If TypeOf row Is GridViewDataRowInfo Then
            Return GetType(ColorSwatchCellElement)
        End If
        Return MyBase.GetCellType(row)
    End Function
End Class

